packages/frontend-web/src/app/util/dataService.ts

```diff
@@ -878,19 +888,90 @@ export async function listAuthorCounts(
 let ws: WebSocket = null;
 let intervalTimer: NodeJS.Timer;
 
+export interface ISystemSummary {
+  tags: List<ITagModel>;
+  taggingSensitivities: List<ITaggingSensitivityModel>;
+  rules: List<IRuleModel>;
+  preselects: List<IPreselectModel>;
+}
+
 export interface IGlobalSummary {
-  deferred: number;
   users: List<IUserModel>;
   categories: List<ICategoryModel>;
   articles: List<IArticleModel>;
+  deferred: number;
 }
 
 export interface IUserSummary {
   assignments: number;
 }
 
+// TODO: API sending number IDs, but we expect strings due to the way the old REST code works.
+//       Convert for now.  But at some point need to refactor to use numbers.
+function packSystemData(data: any): ISystemSummary {
+  return {
+    tags: List<ITagModel>(data.tags.map((t: any) => {
+      t.id = t.id.toString();
+      return TagModel(t);
+    })),
+    taggingSensitivities: List<ITaggingSensitivityModel>(data.taggingSensitivities.map((t: any) => {
+      return TaggingSensitivityModel(t);
+    })),
+    rules: List<IRuleModel>(data.rules.map((r: any) => {
+      return RuleModel(r);
+    })),
+    preselects: List<IPreselectModel>(data.preselects.map((p: any) => {
+      return PreselectModel(p);
+    })),
+  };
+}
+
+function packGlobalData(data: any): IGlobalSummary {
+  const userMap: {[key: number]: IUserModel} = {};
+  const catMap: {[key: number]: ICategoryModel} = {};
+
+  function convertAssignedModerators(am: Array<any>): Array<IUserModel> {
+    return am.map((i) => userMap[i.moderator_assignment.userId]);
+  }
+
+  return {
+    users: List<IUserModel>(data.users.map((u: any) => {
```

Comment:
The input argument is data:any, but it seems that some properties of data are required, like it has a Users array. Can we make the typing of data input argument honest? 

Or if not, explain why in a comment.

Same goes for "any"s below.

Reply:
It would be ideal to have a shared library or some sort of shared type file that specifies the types of the data sent over the wire.  Then we get the type checking on both ends plus free documentation:-)  

I'll have a think about the best way to implement this.  In any case, it'll be a larger change than will fit in a review markup.  I'll add a TODO for now and create a separate change in the next maintenance batch.